src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1138,6 +1138,56 @@ public function getUriForPath($path)
     }
 
     /**
+     * Returns the path as relative reference from the current Request path.
+     *
+     * Only the URIs path component (no schema, host etc.) is relevant and must be given, starting with a slash.
+     * Both paths must be absolute and not contain relative parts.
+     * Relative URLs from one resource to another are useful when generating self-contained downloadable document archives.
+     * Furthermore, they can be used to reduce the link size in documents.
+     *
+     * Example target paths, given a base path of "/a/b/c/d":
+     * - "/a/b/c/d"     -> ""
+     * - "/a/b/c/"      -> "./"
+     * - "/a/b/"        -> "../"
+     * - "/a/b/c/other" -> "other"
+     * - "/a/x/y"       -> "../../x/y"
+     *
+     * @param string $path The target path
+     *
+     * @return string The relative target path
+     */
+    public function getRelativeUriForPath($path)
```

Comment:
@fabpot From what I see Routing [depends](https://github.com/symfony/symfony/blob/2.7/src/Symfony/Component/Routing/composer.json#L23) on HttpFoundation, so moving instead of duplicating code have sense IMO.